upmed-api/src/api/patient/patient.py

```diff
@@ -1,7 +1,294 @@
+            resp = {
+                "googleId": pid,
+                "token": auth_token.decode()
+            }
+            # resp = res
+            # print(resp)
+            return jsonify(resp), 200
+        else:
+            return "False", 404
+    except Exception as e:
+        return f"An Error Occured: {e}"
+
+
+@patient_endpoints.route('/signUp', methods=['POST'])
+def signup():
+    post_data = request.get_json()
+    try:
+        patient = Patient(
+            id= post_data.get('id'),
+            firstName = post_data.get('firstName'),
+            lastName = post_data.get('lastName'),
+            phone = post_data.get('phone'),
+            email = post_data.get('email'),
+            dateOfBirth = post_data.get('dateOfBirth'),
+            sex = post_data.get('sex'),
+            profilePicture = '',
+            height = post_data.get('height'),
+            weight = post_data.get('weight'),
+            drinker = post_data.get('drinker'),
+            smoker = post_data.get('smoker'),
+            calendar=[],
+            health=[],
+            doctors=[]
+        )
+        try:
+            patient.profilePicture = post_data.get('profilePicture')
+        except KeyError:
+            patient.profilePicture = 'https://www.flaticon.com/svg/static/icons/svg/147/147144.svg'
+
+        utype = "PATIENT"
+        pat.document(patient.id).set({
+            "id": patient.id,
+            "firstName": patient.firstName,
+            "lastName": patient.lastName,
+            "phone": patient.phone,
+            "email": patient.email,
+            "dateOfBirth": patient.dateOfBirth,
+            "sex": patient.sex,
+            "profilePicture": patient.profilePicture,
+            "height": patient.height,
+            "weight": patient.weight,
+            "drinker": patient.drinker,
+            "smoker": patient.smoker,
+            "calendar": patient.calendar,
+            "health": patient.health,
+            "doctors": patient.doctors
+
+        })
+        auth_token = auth.encode_auth_token(patient.id, utype)
+        responseObject = {
+            'id': patient.id,
+            'auth_token': auth_token.decode()
+        }
+        return make_response(jsonify(responseObject)), 201
+    except Exception as e:
+        responseObject = {
+            'status': 'fail',
+            'message': f'Some error, {e} occurred. Please try again.'
+        }
+        return make_response(jsonify(responseObject)), 401
+
+@patient_endpoints.route('/delete', methods=['POST'])
+def remove():
+    """
+        delete() : Delete a document from Firestore collection.
+    """
+    post_data = request.get_json()
+    try:
+        # Check for ID in URL query
+        pid = post_data.get('id')
+        pat.document(pid).delete()
+        return jsonify({"success": True}), 200
+    except Exception as e:
+        return f"An Error Occured: {e}"
+
+@patient_endpoints.route('/getByToken', methods=['POST'])
+def getbytoken():
```

Comment:
request body contains the token, no headers

you can just try-except decoding the token